receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go

```diff
@@ -229,6 +235,25 @@ func (s *scraper) scrapeAndAppendDiskIOMetric(now pcommon.Timestamp, handle proc
 	return nil
 }
 
+func (s *scraper) scrapeAndAppendSignalsPendingMetric(now pcommon.Timestamp, handle processHandle) error {
+	if !s.config.Metrics.ProcessSignalsPending.Enabled {
+		return nil
+	}
+
+	rlimitStats, err := handle.RlimitUsage(true)
+	if err != nil {
+		return err
+	}
+
+	for _, rlimitStat := range rlimitStats {
+		if rlimitStat.Resource == process.RLIMIT_SIGPENDING {
+			s.mb.RecordProcessSignalsPendingDataPoint(now, int64(rlimitStat.Used))
```

Comment:
Should we add `break` statement here?

Reply:
Good point, no reason to continue the loop.